enable ninja tests

cfe-NI opened the pull request:
<!--- The title of the PR should summarize the change implemented. -->
<!--- Example commit message format: -->
<!--- `module: summary of change` -->
<!--- (leave blank) -->
<!--- `details of what/why/how an issue was addressed` -->
<!--- Keep subject lines to 50 characters (but 72 is a hard limit!) -->
<!--- characters. Refer to the [Revision Control Guidelines](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md#revision-control-guidelines) section of the coding guidelines -->

## Description
<!--- Provide a general summary of your changes in the title above -->
<!--- Why is this change required? What problem does it solve? -->

## Related Issue
<!--- Refer to any related issues here -->
<!--- If this PR fully addresses an issue, please say "Fixes #1234", -->
<!--- as this will allow Github to automatically close the related Issue -->

## Which blocks/areas does this affect?
<!--- Include blocks that are affected and some details on what -->
<!--- areas these changes affect, such as performance. -->

## Testing Done
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you -->
<!--- ran to see how your change affects other areas of the code, -->
<!--- etc. Then, include justifications for how your tests -->
<!--- demonstrate those affects. -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the
<!--- boxes that apply. Note that some of these may not be valid -->
<!--- for all PRs. -->

- [ ] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [ ] I have squashed my commits to have one significant change per commit. 
- [ ] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [ ] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

mkoop-ni reviewed (changes requested):
Please add signed-off message.

mkoop-ni commented on .github/workflows/win-installer-build.yml:
```diff
@@ -554,12 +554,21 @@ jobs:
+        "%GR_INSTALL_DIR%\python3.12\python.exe" -m pip install pytest
+        cd gnuradio\build
+        @echo Run Ninja tests on installer environment
+        Ninja test
```
Correct, cpack is using the install rules (also used by 'make install'), but does NOT use the content of "$CMAKE_INSTALL_PREFIX".

mkoop-ni commented on gnuradio-runtime/python/pmt/qa_pmt.py:
```diff
@@ -8,6 +8,13 @@
 #
 #
 
+# Add GNU Radio bin directory to DLL search path for Windows
```
Could this be added to a cmake preprocessor variable?

mkoop-ni replied:
I think @Abhverma-NI had a similar problem and even introduced a new variable GR_PKG_DIR using [setupGrPkgPath.cmake](https://github.com/mkoop-ni/gnuradio/pull/22/files#diff-6ae47279a36bf8f6c53b1629631714c362dd3459518e83d91ca23ba7f52ad150). In the ideal world we use a cmake pre-processor variable to insert this macro.

cfe-NI commented:
@mkoop-ni: yesterday we talked about squashing commits. My proposal would be to squash 
eb980042bf582d609c09cf379652883b2328ee57
3f04bb70eb6336dc530f00d788c3445629e61a5b
efc927350a0648c90a3f5e97e662d3a272c65add

since they are all related to cmake logic. do you agree?

cfe-NI commented on gr-utils/read_file_metadata/gr_read_file_metadata:
```diff
@@ -12,6 +12,14 @@ import sys
+    @GR_PKG_DIR_SETUP@
+    gr_bin_dir = os.path.join(os.environ.get('GR_PKG_DIR', ''), 'bin')
+    if os.path.exists(gr_bin_dir):
+        os.add_dll_directory(gr_bin_dir)
```
don't care here, since this is a cherry-pick form another PR

cfe-NI commented:
@mkoop-ni , @Abhverma-NI : I've addressed all comments so far. Please let me know if you would like to change something, otherwise, we can merge this to base-installer-support

mkoop-ni commented on .github/workflows/win-installer-build.yml:
```diff
@@ -549,16 +551,32 @@ jobs:
+    - name: Running Ninja test
+      run: |
+        call "%msbuild_vcvars64%"
+        call %PY_VENV_PATH%\Scripts\activate
```
Wondering if this is even necessary.

mkoop-ni commented on cmake/Modules/GrBoost.cmake:
```diff
@@ -93,7 +93,9 @@ set(Boost_ADDITIONAL_VERSIONS
-    "1.71")
+    "1.71"
+    "1.83.0"
+    "1.83")
```
You listed 1.83.0, since that is the one we are building with on Windows. But shouldn't the list include all to date released versions?

cfe-NI replied:
Yes, will update the list up to the latest release which is 1.89 to allow building with other boost versions.  Releases older than 1.53 I will not add.
(Btw. UHD does not maintain such a list)
